### DISCUSSION TITLE
Octree to polygonal mesh, is that possible?

### DISCUSSION THREAD

**Discussion by FrankEscobar** [libigl/libigl#2086] [Q&A]
If I have a properly conctructed Octree is there any way to dump it to a polygonal mesh? Obviously it could be preferred if internal faces were removed.

Thank you!

**Comment by alecjacobson** [comment]
This isn't implemented yet. Do you want the square faces of the octree? Or something like a marchingcubes/dual contouring of a function stored in a octree?

**Comment by FrankEscobar** [comment]
My intention is to create an octree and visualize it as a mesh, both options creating the square faces or (preferably) marchingcubes/dual contouring could be aceptable.

Thank you for your time!

**Reply by alecjacobson** [reply]
got it. We don't have this implemented. Outputting the square faces should be too bad. An easy to implement solution would be to dump out all 6 faces and 8 vertices for every cell and then eliminate duplicates after the fact.
Meanwhile, if I were going to write a mesh contouring function for values on an octree, I would probably choose dual contouring. I think this is more straightforward than MC for adaptive grids (but admittedly never implemented either on octree myself).
